src/stactools/core/utils/raster_footprint.py

```diff
@@ -26,69 +645,63 @@ def update_geometry_from_asset_footprint(
-            the polygon.
-            A factor of 2 would double the density of points (placing one new point between each
-            existing pair
-            of points), a factor of 3 would place two points between each point, etc.
-        simplify_tolerance (Optional[float]): All points in the simplified object will be within
-            the tolerance distance of the original geometry, in degrees.
-        no_data(Optional[int]): explicitly set the no data value if not in image metadata
-        bands (List[int]): The bands to use to compute the footprint. Defaults
-            to [1]. If an empty list is provided, the bands will be ORd together;
-            e.g. for a pixel to be outside of the footprint, all bands must have
-            nodata in that pixel.
+            extract footprints. The first successful footprint will be used. If
+            the list is empty, all assets will be tried until one is successful.
+        precision (int): The number of decimal places to include in the final
+            footprint coordinates.
+        densification_factor (Optional[int]): The factor by which to
+            increase point density within the polygon before projection to
+            EPSG 4326. A factor of 2 would double the density of points (placing
+            one new point between each existing pair of points), a factor of
+            3 would place two points between each point, etc. Higher
+            densities produce higher fidelity footprints in areas of high
+            projection distortion.
+        simplify_tolerance (Optional[float]): Distance, in degrees, within which
+            all locations on the simplified polygon will be to the original
+            polygon.
+        no_data (Optional[Union[int, float]]): Explicitly sets the nodata value
+            if not in source image metadata. If set to None, this will return
+            a footprint including nodata values.
+        bands (List[int]): The bands to use to compute the footprint.
+            Defaults to [1]. If an empty list is provided, the bands will be ORd
+            together; e.g., for a pixel to be outside of the footprint, all
+            bands must have nodata in that pixel.
+        skip_errors (bool): If False, raise an error for a missing href or
+            footprint calculation failure.
 
     Returns:
-        bool: True if the extent was successfully updated, False if not
+        bool: True if the Item geometry was successfully updated, False if not.
     """
-    asset_name_and_extent = next(
-        data_footprints_for_data_assets(
-            item,
-            asset_names=asset_names,
-            precision=precision,
-            densification_factor=densification_factor,
-            simplify_tolerance=simplify_tolerance,
-            no_data=no_data,
-            bands=bands,
-        ),
-        None,
+    return RasterFootprint.update_geometry_from_asset_footprint(
```

Comment:
This should also use `warnings.warn` to programatically deprecate the function. Same is true for the other depreacated free functions.

Reply:
Done.